File: java/program/Laptop.java
public class Laptop extends Elektronik {
    private String merek;
    private String warna;
    private String processor;

    // Constructor default
    public Laptop() {
        super();
        this.merek = "";
        this.warna = "";
        this.processor = "";
    }

    // Constructor dengan parameter
    public Laptop(String id, String namaProduk, int harga,
                  String kategori, int jumlahStok, String statusBarang,
                  String merek, String warna, String processor) {
        super(id, namaProduk, harga, kategori, jumlahStok, statusBarang); // panggil parent (Elektronik)
        this.merek = merek;
        this.warna = warna;
        this.processor = processor;
    }

    // Getter & Setter
    public String getMerek() {
        return merek;
    }

    public void setMerek(String merek) {
        this.merek = merek;
    }

    public String getWarna() {
        return warna;
    }

    public void setWarna(String warna) {
        this.warna = warna;
    }

    public String getProcessor() {
        return processor;
    }

    public void setProcessor(String processor) {
        this.processor = processor;
    }
}

File: cpp/program/Laptop.cpp
#include <iostream>
#include "Elektronik.cpp"
using namespace std;

// Class Laptop (Child) dari class Elektronik (Parent)
class Laptop : public Elektronik {
private:
    string merek; // Atribut merek
    string warna; // Atribut warna
    string processor; // Atribut processor

    public:
    // Constructor kosong
    Laptop() {
    }

    // Constructor dengan parameter
    Laptop(string id, string namaProduk, int harga, string kategori, int jumlahStok, string statusBarang,
           string merek, string warna, string processor)
        : Elektronik(id, namaProduk, harga, kategori, jumlahStok, statusBarang) {
        this->merek = merek;
        this->warna = warna;
        this->processor = processor;
    }

    // Getter setter
    string getMerek() const {
        return merek;
    }
    void setMerek(string merek) {
        this->merek = merek;
    }

    string getWarna() const {
        return warna;
    }
    void setWarna(string warna) {
        this->warna = warna;
    }

    string getProcessor() const {
        return processor;
    }
    void setProcessor(string processor) {
        this->processor = processor;
    }
};
